final project idea

tamanduaa opened the pull request.

tamanduaa committed:
final project idea

jamdon commented:
Hi Lauren,

Only you can know how difficult it might be as I'm not familiar with the framework you're going to use. My main advice is to think now in terms of how you'll incorporate the required elements of the project.

tamanduaa committed:
updated project idea

tamanduaa committed:
first commit

jamdon commented:
Hi Lauren,

Just wanted to clarify if this is your final submission and you want it graded, if you want feedback on it so you can work on it more, or if you have more work incoming later today.

tamanduaa committed:
update

tamanduaa commented:
Hi James,

I would love feedback on this if you can.  I'm still working on it but I am
a bit stuck so any feedback or direction on how to properly set up backbone
would be great.  I am also trying to connect it so that the links at the
bottom of the page link up to my props list not the food trucks list.  I
have just made another commit.

Thank you so much!
Lauren

On Sat, Jun 18, 2016 at 12:19 PM, jamdon notifications@github.com wrote:

> Hi Lauren,
> 
> Just wanted to clarify if this is your final submission and you want it
> graded, if you want feedback on it so you can work on it more, or if you
> have more work incoming later today.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226960214,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-PkuW9oLrP1zO6YmT8u8a6trzYv70ks5qNETDgaJpZM4ITggw
> .

jamdon commented:
The first thing you need is to add the proper dev dependencies to your package.json file. If you're going to use grunt, then you have to install (and save to your package.json file in the dev dependencies) the grunt plugins that the Gruntfile is specifying so it can build your client script bundles.

tamanduaa committed:
added dependencies for grunt

tamanduaa commented:
Thank you so much James! I have installed the grunt plugins. I am trying to
add a link at the bottom that will work the way that the food trucks
example sorts trucks by date or type.  But in this example it would sort my
list of props by material and due date.  I am a bit confused about how to
make those links reference my list.  I have been trying to work on the
Props List one.  It goes to the right URL but displays nothing.  How can I
connect it to the right data?

Thank you,
Lauren

On Sat, Jun 18, 2016 at 12:46 PM, jamdon notifications@github.com wrote:

> The first thing you need is to add the proper dev dependencies to your
> package.json file. If you're going to use grunt, then you have to install
> (and save to your package.json file in the dev dependencies) the grunt
> plugins that the Gruntfile is specifying so it can build your client script
> bundles.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226961438,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-PhOFlA_lhrUlvsTnZA7jjkpkyv5Cks5qNEsXgaJpZM4ITggw
> .

jamdon commented on index.js:
```diff
@@ -0,0 +1,38 @@
+app.use('/today', foodTypeRouter);
+app.use('/props', foodTypeRouter);
+
+app.use(express.static(root));
```
If you're going to use root as a variable here, then you have to define it first.

jamdon commented:
Ok, I was able to compile your client-side files, but I can't run your server yet. I left a comment inline.

tamanduaa committed:
defined root

tamanduaa commented:
Thank you James!  I have defined root and commented out the line:

app.use(express.static('public'));

because I'm guessing that conflicts with

app.use(express.static(root));

Are you able to run the server now?

Thank you!

Lauren

On Sat, Jun 18, 2016 at 2:21 PM, jamdon notifications@github.com wrote:

> Ok, I was able to compile your client-side files, but I can't run your
> server yet. I left a comment inline.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226965574,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-Pu3nVNfEbKJljv_d8aHyTnW-hQiwks5qNGFOgaJpZM4ITggw
> .

jamdon commented:
You're not getting any data back when you try to fetch in client.js because you're making a call to the /trucks endpiont, but your express router is mounted on the /api/trucks route. Also, just so I'm clear, it doesn't appear you actually use any Backbone components in your client side application. Is that correct?

tamanduaa committed:
updated path to call the trucks endpoint rather than the api/trucks endpoint.  updated nav links to go to correct url

tamanduaa commented:
Thank you James!  I've updated it to make a call to /trucks instead and
changed the links so it appears they are going to the correct url now.  I'm
in the process of updating the view files.  Once I add view files that
display my props list and filter it, I think that would mean I am using
Backbone components in my client side app.  Am I correct in thinking that?

Thank you,
Lauren

On Sat, Jun 18, 2016 at 3:35 PM, jamdon notifications@github.com wrote:

> You're not getting any data back when you try to fetch in client.js
> because you're making a call to the /trucks endpiont, but your express
> router is mounted on the /api/trucks route. Also, just so I'm clear, it
> doesn't appear you actually use any Backbone components in your client side
> application. Is that correct?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226968812,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-PjKWfQYrUttqIz3_IqPIavs8p_Dzks5qNHKZgaJpZM4ITggw
> .

tamanduaa committed:
updated controller and added prop views

tamanduaa committed:
updated templates and views

tamanduaa committed:
update

jamdon commented:
I see the files, but nothing is ever used. Your index.html never loads your main.bundled.js or your templates.js. Was it your intention to use any of it?

tamanduaa commented:
Hi James,

Ahh okay I see that.  I've added them to index.html.  My intention was to
use templates to write out a filtered list of props, but I ran out of time
to write the code to filter them.  I wanted to have a button to show all
props that used a certain material.  So I would display a list of the props
whose entry for that material was greater than zero.  Ideally that list
would show the prop name and the material amounts, and I thought
templates.js would be needed to do that.

Thank you,
Lauren

On Sat, Jun 18, 2016 at 7:11 PM, jamdon notifications@github.com wrote:

> I see the files, but nothing is ever used. Your index.html never loads
> your main.bundled.js or your templates.js. Was it your intention to use any
> of it?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226975262,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-PjKztu11Ab9SzQlzU0NfZeiga1itks5qNKU7gaJpZM4ITggw
> .

tamanduaa committed:
added links to index

tamanduaa commented:
Also I realize time is up because the project is due today.  In order to
submit it do I need to do a pull request?

Thank you,
Lauren

On Sat, Jun 18, 2016 at 7:28 PM, Lauren Perry laurellperry@gmail.com
wrote:

> Hi James,
> 
> Ahh okay I see that.  I've added them to index.html.  My intention was to
> use templates to write out a filtered list of props, but I ran out of time
> to write the code to filter them.  I wanted to have a button to show all
> props that used a certain material.  So I would display a list of the props
> whose entry for that material was greater than zero.  Ideally that list
> would show the prop name and the material amounts, and I thought
> templates.js would be needed to do that.
> 
> Thank you,
> Lauren
> 
> On Sat, Jun 18, 2016 at 7:11 PM, jamdon notifications@github.com wrote:
> 
> > I see the files, but nothing is ever used. Your index.html never loads
> > your main.bundled.js or your templates.js. Was it your intention to use any
> > of it?
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226975262,
> > or mute the thread
> > https://github.com/notifications/unsubscribe/ACo-PjKztu11Ab9SzQlzU0NfZeiga1itks5qNKU7gaJpZM4ITggw
> > .

jamdon commented:
No, this is your pull request. You've just been updating it. 👍

tamanduaa commented:
Got it, thank you so much for all your help James!

On Sat, Jun 18, 2016 at 8:35 PM, jamdon notifications@github.com wrote:

> No, this is your pull request. You've just been updating it. 👍
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Final_Project/pull/9#issuecomment-226977490,
> or mute the thread
> https://github.com/notifications/unsubscribe/ACo-PtMHOhQZgnL5GKN5wvgJs6aR5WTFks5qNLkRgaJpZM4ITggw
> .
